src/NServiceBus.Transport.SQS/MessageDispatcher.cs

```diff
@@ -363,8 +363,16 @@ async Task ApplyUnicastOperationMappingIfNecessary(UnicastTransportOperation tra
             else
             {
                 sqsPreparedMessage.Destination = transportOperation.Destination;
-                sqsPreparedMessage.QueueUrl = await queueCache.GetQueueUrl(sqsPreparedMessage.Destination)
-                    .ConfigureAwait(false);
+
+                try
+                {
+                    sqsPreparedMessage.QueueUrl = await queueCache.GetQueueUrl(sqsPreparedMessage.Destination)
+                        .ConfigureAwait(false);
+                }
+                catch (QueueDoesNotExistException ex)
```

Comment:
We can look into this as a dedicated PR. I'm just bringing this up for now. Core has also the [`QueueNotFoundException` type](https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/Queuing/QueueNotFoundException.cs) that we could use instead

https://github.com/search?q=org%3AParticular+QueueNotFoundException&type=code

Reply:
We used the QueueDoesNotExistException as that is what is already used in v6 of the transport. It was only v5 that wasn't catching this.
So if we change it, we should be changing it in all the other places that v6 is using it for consistency.

Reply:
And changing it might be a breaking change